File: algo-method.com/tasks/1102/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];
const print = (...messages: any) => console.log(...messages);

const N = +input();
const A = input().split(" ").map(Number);

let currentMax = 0;
let currentMaxCount = 0;

const ans = A.map((a) => {
  if (a === currentMax) {
    currentMaxCount++;
  } else if (currentMax < a) {
    currentMax = a;
    currentMaxCount = 1;
  }
  return currentMaxCount;
});

print(ans.join("\n"));

File: algo-method.com/tasks/1102/main.py
N = int(input())
A = list(map(int, input().split()))

current_max = 0
current_max_cnt = 0
ans: "list[int]" = []

for a in A:
    if a == current_max:
        current_max_cnt += 1
    elif current_max < a:
        current_max = a
        current_max_cnt = 1
    ans.append(current_max_cnt)

print(*ans, sep="\n")
